pallets/schemas/src/lib.rs

```diff
@@ -381,11 +445,204 @@ pub mod pallet {
+		#[pallet::weight(
+			match schema_name {
+				Some(_) => T::WeightInfo::propose_to_create_schema_v2(model.len() as u32),
+				None => T::WeightInfo::propose_to_create_schema(model.len() as u32)
+			}
+		)]
+		pub fn propose_to_create_schema_v2(
+			origin: OriginFor<T>,
+			model: BoundedVec<u8, T::SchemaModelMaxBytesBoundedVecLimit>,
+			model_type: ModelType,
+			payload_location: PayloadLocation,
+			settings: BoundedVec<SchemaSetting, T::MaxSchemaSettingsPerSchema>,
+			schema_name: Option<SchemaNamePayload>,
+		) -> DispatchResult {
+			let proposer = ensure_signed(origin)?;
+
+			let proposal: Box<T::Proposal> = Box::new(
+				(Call::<T>::create_schema_via_governance_v2 {
+					creator_key: proposer.clone(),
+					model,
+					model_type,
+					payload_location,
+					settings,
+					schema_name,
+				})
+				.into(),
+			);
+			T::ProposalProvider::propose_with_simple_majority(proposer, proposal)?;
+			Ok(())
+		}
+
+		/// Create a schema by means of council approval
+		///
+		/// # Events
+		/// * [`Event::SchemaCreated`]
+		///
+		/// # Errors
+		/// * [`Error::LessThanMinSchemaModelBytes`] - The schema's length is less than the minimum schema length
+		/// * [`Error::ExceedsMaxSchemaModelBytes`] - The schema's length is greater than the maximum schema length
+		/// * [`Error::InvalidSchema`] - Schema is malformed in some way
+		/// * [`Error::SchemaCountOverflow`] - The schema count has exceeded its bounds
+		#[pallet::call_index(6)]
+		#[pallet::weight(
+			match schema_name {
+				Some(_) => T::WeightInfo::create_schema_via_governance_v2(model.len() as u32+ settings.len() as u32),
+				None => T::WeightInfo::create_schema_via_governance(model.len() as u32+ settings.len() as u32)
+			}
+		)]
+		pub fn create_schema_via_governance_v2(
+			origin: OriginFor<T>,
+			creator_key: T::AccountId,
+			model: BoundedVec<u8, T::SchemaModelMaxBytesBoundedVecLimit>,
+			model_type: ModelType,
+			payload_location: PayloadLocation,
+			settings: BoundedVec<SchemaSetting, T::MaxSchemaSettingsPerSchema>,
+			schema_name: Option<SchemaNamePayload>,
+		) -> DispatchResult {
+			T::CreateSchemaViaGovernanceOrigin::ensure_origin(origin)?;
+			let (schema_id, schema_name) = Self::create_schema_for(
+				model,
+				model_type,
+				payload_location,
+				settings,
+				schema_name,
+			)?;
+
+			Self::deposit_event(Event::SchemaCreated { key: creator_key, schema_id });
+			if let Some(inner_name) = schema_name {
+				Self::deposit_event(Event::SchemaNameCreated {
+					schema_id,
+					name: inner_name.get_combined_name(),
+				});
+			}
+			Ok(())
+		}
+
+		/// Adds a given schema to storage. The schema in question must be of length
+		/// between the min and max model size allowed for schemas (see pallet
+		/// constants above). If the pallet's maximum schema limit has been
+		/// fulfilled by the time this extrinsic is called, a SchemaCountOverflow error
+		/// will be thrown.
+		///
+		/// # Events
+		/// * [`Event::SchemaCreated`]
```

Comment:
Same here. SchemaNameCreated event is missing.

Reply:
Updated